### ISSUE TITLE
Results grouping can effectively increase priority of a fail

### ISSUE THREAD

**Issue by mhidas** [ioos/compliance-checker#103]
For example, in [this file](http://thredds.aodn.org.au/thredds/catalog/IMOS/eMII/checker_test/ANMN/timeSeries/catalog.html?dataset=IMOS/eMII/checker_test/ANMN/timeSeries/IMOS_ANMN-AM_GST_20140407T040000Z_NRSMAI_FV01_NRSMAI-CO2-1404-delayed_END_20140922T220000Z_C-20150311T084035Z.acceptable_units.nc) (with `LATITUDE:units = "degree_north"`) the only failed checks are for latitude and longitude units:

```
latitude                               :3:     3/ 4 :  
    LATITUDE                           :3:     3/ 4 :  
        correct_units                  :2:     2/ 3 : LATITUDE units are
                                                      acceptable, but not
                                                      recommended
```

The check that failed is MEDIUM priority, but because of the way result values are aggregated, it makes it look like a HIGH priority fail. So even if I think "acceptable" units are good enough and specify `--criteria=lenient` on the command line, this file will still fail the CF checks.

**Comment by mhidas** [comment]
I don't know yet how many checks are affected by this issue, but it could potentially be an annoying problem for the way we'd like to use this checker at IMOS (see https://github.com/ioos/compliance-checker/issues/37#issuecomment-93616335).

Suggested solutions:
* Only group together checks of the same priority level; or
* When deciding on the final PASS/FAIL result, only consider the value & priority of each individual check, not the aggregated score & top-level priority of each group.

**Comment by lukecampbell** [comment]
I can take a look at the file if you could provide an updated URL for it. As far as regrouping goes, the grouping and aggregations are based on the language used in the documents. For example when ACDD uses the language "highly recommended" the check counts as a high priority or 3 points. When ACDD uses language like "recommended" it's a medium check. 

I'll take regrouping into consideration.

**Comment by mhidas** [comment]
@lukecampbell Sorry, we've made some  changes to our THREDDS server. [Here](http://thredds.aodn.org.au/thredds/catalog/IMOS/eMII/checker_test/ANMN/timeSeries/catalog.html?dataset=eMII/checker_test/ANMN/timeSeries/IMOS_ANMN-AM_GST_20140407T040000Z_NRSMAI_FV01_NRSMAI-CO2-1404-delayed_END_20140922T220000Z_C-20150311T084035Z.acceptable_units.nc) is an updated link to the file mentioned above.

**Comment by daf** [comment]
This specific file no longer demonstrates the issue - was likely addressed in #167 or similar as things started getting ungrouped.  Current output:

```
                  Reasoning for the failed tests given below:                   


Name                             Priority:     Score:Reasoning
--------------------------------------------------------------------------------
4.1 Coordinates representing latitude  :2:     2/ 3 : LATITUDE units are
                                                      acceptable, but not
                                                      recommended
4.1 Coordinates representing longitude :2:     2/ 3 : LONGITUDE units are
                                                      acceptable, but not
                                                      recommended
```

However, the underlying issue has not been addressed - grouping will always take the highest priority of failing children and promote that to the group's priority.  I'm not sure this is even a bug, as long as it is clear on reporting, but as we've talked about moving away from the grouping idea (see #37), maybe this just goes away?

**Comment by mhidas** [comment]
@daf Yes, if the final PASS/FAIL result is determined by considering each individual check with its own priority level independently, then yes, this issue would go away.

However, when presenting the results, I think it is helpful to group the failed tests in a meaningful way (e.g. group all checks for a given variable together, like it was before #167 -- I have proposed a solution in #194).

**Comment by lukecampbell** [comment]
I'm slowly refactoring out the score groups.

**Comment by benjwadams** [comment]
Has been addressed in ACDD by #561 by breaking out groups/issues into separate entries with corresponding importances.  As of now, it's still possible to group, but none of the core checkers are using grouping and it should be considered deprecated behavior.  I'm going to close this issue with the caveat that grouping still exists, but should not be used going forward.

**Comment by mhidas** [comment]
> As of now, it's still possible to group, but none of the core checkers are using grouping and it should be considered deprecated behavior.

So is there a plan to remove the somewhat complicated grouping code, and just report all the results as is? (could still be _sorted_ by name, just not grouped)

After the changes to the text output format in v4.0 I re-named all the results returned in [our plugin](https://github.com/aodn/cc-plugin-imos) from tuples to simple strings. Then we ran into this grouping problem again, i.e. a low-priority check suddenly became high priority just because it had the same name (== the variable name) as a high-priority one. Sure I can get around it by re-naming the results, but if grouping is deprecated, I shouldn't need to play tricks just to avoid doing it.

Also, I used to think I could get around this by extracting the "raw" results in the json output format, but it seems the grouping is applied to that output as well.
